client.go

```diff
@@ -1462,22 +1462,22 @@ func (c *HostClient) doNonNilReqResp(req *Request, resp *Response) (bool, error)
 	}
 
 	br := c.acquireReader(conn)
-	if err = resp.ReadLimitBody(br, c.MaxResponseBodySize); err != nil {
-		c.releaseReader(br)
+	err = resp.ReadLimitBody(br, c.MaxResponseBodySize)
+	c.releaseReader(br)
+	if err != nil {
 		c.closeConn(cc)
 		// Don't retry in case of ErrBodyTooLarge since we will just get the same again.
 		retry := err != ErrBodyTooLarge
 		return retry, err
 	}
-	c.releaseReader(br)
 
 	if resetConnection || req.ConnectionClose() || resp.ConnectionClose() {
 		c.closeConn(cc)
 	} else {
 		c.releaseConn(cc)
```

Comment:
Now when a connection is closed with `ECONNRESET` it will still be released to the pool here (as there was no error before) and will try to be used again.

Reply:
```if resetConnection || req.ConnectionClose() || resp.ConnectionClose() || isECONNRESET```

Reply:
@moredure true. Did not see this case. I did make the change as @moredure suggested. Thanks!

Reply:
Ack